src/socket.ts

```diff
@@ -319,7 +328,13 @@ export class ConstellationSocket extends EventEmitter {
     }
 
     private rebroadcastEvent(name: string) {
-        this.socket.addEventListener(name, evt => this.emit(name, evt));
+        const socket = this.socket;
+        socket.addEventListener(name, evt => {
+            // Sanity check that the socket is current.
+            if (this.socket === socket) {
```

Comment:
We'd probably make a `Reconnecting` state, since `Closing` is used when the client wants to close without reconnection. But essentially yes.